django/db/migrations/autodetector.py

```diff
@@ -867,6 +867,13 @@ def generate_altered_fields(self):
                 )
                 if rename_key in self.renamed_models:
                     new_field.remote_field.model = old_field.remote_field.model
+            if hasattr(new_field, "remote_field") and getattr(new_field.remote_field, "through", None):
+                rename_key = (
+                    new_field.remote_field.through._meta.app_label,
+                    new_field.remote_field.through._meta.model_name,
+                )
+                if rename_key in self.renamed_models:
+                    new_field.remote_field.through = old_field.remote_field.through
```

Comment:
I think this might be a bit clearer:

```python
try:
    meta = new_field.remote_field.through._meta
except AttributeError:
    pass
else:
    rename_key = (meta.app_label, meta.model_name)
    if rename_key in self.renamed_models:
        new_field.remote_field.through = old_field.remote_field.through
```